sky/provision/kubernetes/instance.py

```diff
@@ -222,6 +222,28 @@ def _wait_for_pods_to_run(namespace, new_nodes):
     Pods may be pulling images or may be in the process of container
     creation.
     """
+
+    def _check_init_containers(pod):
+        # Check if any of the init containers failed
+        # to start. Could be because the init container
+        # command failed or failed to pull image etc.
+        for init_status in pod.status.init_container_statuses:
+            init_terminated = init_status.state.terminated
+            if init_terminated and init_terminated.exit_code != 0:
+                msg = init_terminated.message if (
+                    init_terminated.message) else str(init_terminated)
+                raise config_lib.KubernetesError(
+                    f'Failed to run init container for pod {pod.metadata.name}.'
+                    f' Error details: {msg}.')
+            init_waiting = init_status.state.waiting
```

Comment:
nit: do we still need to check `init_waiting` if `init_terminated` is true?

Reply:
Good point, checking `init_waiting` is not required when `init_terminated` is not None. Updated.

Reply:
Oh, should this be `return` or `continue`? Will there be multiple init_containers?

Reply:
Oops good catch - should be `continue`